docs/reference/react-style-guide.md

```diff
@@ -0,0 +1,534 @@
+
+## Spacing
+
+- Always include a single space in your self-closing tag. eslint: `no-multi-spaces`, `react/jsx-tag-spacing`
+
+```javascript
+// bad
+<Foo/>
+
+// very bad
+<Foo                 />
+
+// bad
+<Foo
+ />
+
+// good
+<Foo />
+```
+
+- Do not pad JSX curly braces with spaces. eslint: `react/jsx-curly-spacing`
+
+```javascript
+// bad
+<Foo bar={ baz } />
+
+// good
+<Foo bar={baz} />
+```
+
+## Props
+
+- Always use camelCase for prop names.
+
+```javascript
+// bad
+<Foo
+  UserName="hello"
+  phone_number={12345678}
+/>
+
+// good
+<Foo
+  userName="hello"
+  phoneNumber={12345678}
+/>
+```
+
+- Omit the value of the prop when it is explicitly true. eslint: `react/jsx-boolean-value`
+
+```javascript
+// bad
+<Foo
+  hidden={true}
+/>
+
+// good
+<Foo
+  hidden
+/>
+
+// good
+<Foo hidden />
+```
+
+- Always include an alt prop on `<img>` tags. If the image is presentational, alt can be an empty string or the `<img>` must have role="presentation". eslint: `jsx-a11y/alt-text`
+
+```javascript
+// bad
+<img src="hello.jpg" />
+
+// good
+<img src="hello.jpg" alt="Me waving hello" />
+
+// good
+<img src="hello.jpg" alt="" />
+
+// good
+<img src="hello.jpg" role="presentation" />
+```
+
+- Do not use words like “image”, “photo”, or “picture” in `<img>` alt props. eslint: `jsx-a11y/img-redundant-alt`
+  - Why? Screenreaders already announce img elements as images, so there is no need to include this information in the alt text.
+
+```javascript
+// bad
+<img src="hello.jpg" alt="Picture of me waving hello" />
+
+// good
+<img src="hello.jpg" alt="Me waving hello" />
+```
+
+- Use only valid, non-abstract ARIA roles. eslint: jsx-a11y/aria-role
+
+```javascript
+// bad - not an ARIA role
+<div role="datepicker" />
+
+// bad - abstract ARIA role
+<div role="range" />
+
+// good
+<div role="button" />
+```
+
+- Do not use accessKey on elements. eslint: `jsx-a11y/no-access-key`
+  - Why? Inconsistencies between keyboard shortcuts and keyboard commands used by people using screenreaders and keyboards complicate accessibility.
+
+```javascript
+// bad
+<div accessKey="h" />
+
+// good
+<div />
+```
+
+- Avoid using an array index as key prop, prefer a stable ID. eslint: `react/no-array-index-key`
+  - Why? Not using a stable ID [is an anti-pattern](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318) because it can negatively impact performance and cause issues with component state.
+
+```javascript  
+// bad
+{todos.map((todo, index) =>
+  <Todo
+    {...todo}
+    key={index}
+  />
+)}
+
+// good
+{todos.map(todo => (
+  <Todo
+    {...todo}
+    key={todo.id}
+  />
+))}
+```
+
+- Use spread props sparingly.
+  - Why? Otherwise you’re more likely to pass unnecessary props down to components. And for React v15.6.1 and older, you could [pass invalid HTML attributes to the DOM](https://reactjs.org/blog/2017/09/08/dom-attributes-in-react-16.html).
+
+Exceptions:
+
+- HOCs that proxy down props and hoist propTypes
```

Comment:
I don't think we use `propTypes` as others mentioned in Anna's [draft Style Guide doc](https://docs.google.com/document/d/1x_n0xZHM7oT_Z7hOqMo1Kl7KdICLlVd9hE9vb869mTk/edit?usp=sharing).